api/v1alpha1/promotion_types.go

```diff
@@ -96,6 +96,8 @@ type PromotionStatus struct {
 	// Metadata holds arbitrary metadata set by promotion mechanisms
 	// (e.g. for display purposes, or internal bookkeeping)
 	Metadata map[string]string `json:"metadata,omitempty" protobuf:"bytes,3,rep,name=metadata" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
+	// Freight is the detail of the piece of freight that was referenced by this promotion.
+	Freight *FreightReference `json:"promotedFreight,omitempty" protobuf:"bytes,4,opt,name=freight"`
```

Comment:
You can run `make hack-codegen` which will replace the files.

Reply:
> You can run make hack-codegen which will replace the files.

Sometimes merge conflicts prevent you from easily getting to a spot where you can do this.

I usually isolate all codegen activity to its own commit so that in the even of a merge or rebase, I can remove the old codegen commit ahead of time to prevent conflict with `main` and then a new codegen run should work fine.

But please don't waste a lot of time on that...

I would be happy to straighten out the codegen once everything else about the PR looks right.